crates/typos-cli/src/file.rs

```diff
@@ -664,6 +670,19 @@ fn fix_buffer(mut buffer: Vec<u8>, typos: impl Iterator<Item = typos::Typo<'stat
     buffer
 }
 
+fn fix_file_name<'a>(
```

Comment:
Since you are already going back and making a change, could you split the extraction of `fix_file_name` out into its own commit before the adding of indices?  Each commit should do only one thing.  This is a big help for reviewers as I don't have to try to untangle the root reason for each change